// sql-components/esm2020/lib/sql-card/sql-card.component.mjs
import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class SqlCardComponent {
    constructor() {
        this.title = '';
        this.closed = 'N';
        this.style = 'margin-top:10px; margin-bottom:10px; margin-left: 10px; margin-right: 10px;';
        this.button_class = 'btn btn-primary float-right';
        this.button_style = 'margin-top: 10px; margin-bottom: 10px; text-align: right!important;';
        this.title_class = 'm-2';
        this.title_style = 'font-size: 18px; font-weight: bold;';
    }
    toggleClosed() {
        if (this.closed == 'Y') {
            this.closed = 'N';
        }
        else {
            this.closed = 'Y';
        }
    }
}
SqlCardComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlCardComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
SqlCardComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlCardComponent, isStandalone: true, selector: "sql-card", inputs: { title: "title", closed: "closed", style: "style", button_class: "button_class", button_style: "button_style", title_class: "title_class", title_style: "title_style" }, ngImport: i0, template: "<div class=\"card\" [style]=\"style\">\n    <div class=\"card-header\" style=\"margin-top: 0px;\">\n            <div class=\"col-9 col-lg-10 col-xl-11\">\n                <span [class]=\"title_class\" [style]=\"title_style\">{{ title }}</span>\n                  <span class=\"align-self-end float-end\" style=\"text-align: right;\">\n                    <button (click)=\"toggleClosed()\" *ngIf=\"closed=='Y'\" [class]=\"button_class\" [style]=\"button_style\"><i class=\"bi bi-arrow-bar-down\"></i></button>\n                   <button (click)=\"toggleClosed()\" *ngIf=\"closed=='N'\" [class]=\"button_class\" [style]=\"button_style\"><i class=\"bi bi-arrow-bar-up\"></i></button>\n                </span>\n            </div>\n    </div>\n    <div *ngIf=\"closed=='N'\" class=\"card-body p-2\">\n        <ng-content></ng-content>\n    </div>\n</div>\n", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlCardComponent, decorators: [{
            type: Component,
            args: [{ selector: 'sql-card', standalone: true, imports: [CommonModule], template: "<div class=\"card\" [style]=\"style\">\n    <div class=\"card-header\" style=\"margin-top: 0px;\">\n            <div class=\"col-9 col-lg-10 col-xl-11\">\n                <span [class]=\"title_class\" [style]=\"title_style\">{{ title }}</span>\n                  <span class=\"align-self-end float-end\" style=\"text-align: right;\">\n                    <button (click)=\"toggleClosed()\" *ngIf=\"closed=='Y'\" [class]=\"button_class\" [style]=\"button_style\"><i class=\"bi bi-arrow-bar-down\"></i></button>\n                   <button (click)=\"toggleClosed()\" *ngIf=\"closed=='N'\" [class]=\"button_class\" [style]=\"button_style\"><i class=\"bi bi-arrow-bar-up\"></i></button>\n                </span>\n            </div>\n    </div>\n    <div *ngIf=\"closed=='N'\" class=\"card-body p-2\">\n        <ng-content></ng-content>\n    </div>\n</div>\n" }]
        }], propDecorators: { title: [{
                type: Input
            }], closed: [{
                type: Input
            }], style: [{
                type: Input
            }], button_class: [{
                type: Input
            }], button_style: [{
                type: Input
            }], title_class: [{
                type: Input
            }], title_style: [{
                type: Input
            }] } });
