// sql-components/esm2020/lib/sql-components.module.mjs
import { NgModule } from '@angular/core';
import { SqlComponentsComponent } from './sql-components.component';
import { SqlFormComponent } from './sql-form/sql-form.component';
import { SqlAddTableComponent } from './sql-add-table/sql-add-table.component';
import { SqlCheckboxComponent } from './sql-checkbox/sql-checkbox.component';
import { SqlFullcalendarComponent } from './sql-fullcalendar/sql-fullcalendar.component';
import { SqlPanelComponent } from './sql-panel/sql-panel.component';
import { SqlCardComponent } from './sql-card/sql-card.component';
import { SqlDatepickerComponent } from './sql-datepicker/sql-datepicker.component';
import { SqlUploadFormComponent } from './sql-upload-form/sql-upload-form.component';
import { SqlInputComponent } from './sql-input/sql-input.component';
import { SqlSelectComponent } from './sql-select/sql-select.component';
import { SqlTextareaComponent } from './sql-textarea/sql-textarea.component';
import { SqlCalendarComponent } from './sql-calendar/sql-calendar.component';
import { SqlRadioGroupComponent } from './sql-radio-group/sql-radio-group.component';
import { SqlTableComponent } from './sql-table/sql-table.component';
import { SqlEditTableComponent } from './sql-edit-table/sql-edit-table.component';
import { SqlSubmitComponent } from './sql-submit/sql-submit.component';
import * as i0 from "@angular/core";
export class SqlComponentsModule {
}
SqlComponentsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SqlComponentsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.3", ngImport: i0, type: SqlComponentsModule, declarations: [SqlComponentsComponent], imports: [SqlFormComponent,
        SqlAddTableComponent,
        SqlCalendarComponent,
        SqlCheckboxComponent,
        SqlFullcalendarComponent,
        SqlPanelComponent,
        SqlCardComponent,
        SqlDatepickerComponent,
        SqlUploadFormComponent,
        SqlInputComponent,
        SqlSelectComponent,
        SqlTextareaComponent,
        SqlRadioGroupComponent,
        SqlTableComponent,
        SqlEditTableComponent,
        SqlSubmitComponent], exports: [SqlComponentsComponent,
        SqlFormComponent,
        SqlAddTableComponent,
        SqlCalendarComponent,
        SqlCheckboxComponent,
        SqlFullcalendarComponent,
        SqlPanelComponent,
        SqlCardComponent,
        SqlDatepickerComponent,
        SqlUploadFormComponent,
        SqlInputComponent,
        SqlSelectComponent,
        SqlTextareaComponent,
        SqlRadioGroupComponent,
        SqlTableComponent,
        SqlEditTableComponent,
        SqlSubmitComponent] });
SqlComponentsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlComponentsModule, imports: [SqlFormComponent,
        SqlAddTableComponent,
        SqlCalendarComponent,
        SqlCheckboxComponent,
        SqlFullcalendarComponent,
        SqlPanelComponent,
        SqlCardComponent,
        SqlDatepickerComponent,
        SqlUploadFormComponent,
        SqlInputComponent,
        SqlSelectComponent,
        SqlTextareaComponent,
        SqlRadioGroupComponent,
        SqlTableComponent,
        SqlEditTableComponent,
        SqlSubmitComponent] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        SqlComponentsComponent,
                    ],
                    imports: [
                        SqlFormComponent,
                        SqlAddTableComponent,
                        SqlCalendarComponent,
                        SqlCheckboxComponent,
                        SqlFullcalendarComponent,
                        SqlPanelComponent,
                        SqlCardComponent,
                        SqlDatepickerComponent,
                        SqlUploadFormComponent,
                        SqlInputComponent,
                        SqlSelectComponent,
                        SqlTextareaComponent,
                        SqlRadioGroupComponent,
                        SqlTableComponent,
                        SqlEditTableComponent,
                        SqlSubmitComponent
                    ],
                    exports: [
                        SqlComponentsComponent,
                        SqlFormComponent,
                        SqlAddTableComponent,
                        SqlCalendarComponent,
                        SqlCheckboxComponent,
                        SqlFullcalendarComponent,
                        SqlPanelComponent,
                        SqlCardComponent,
                        SqlDatepickerComponent,
                        SqlUploadFormComponent,
                        SqlInputComponent,
                        SqlSelectComponent,
                        SqlTextareaComponent,
                        SqlRadioGroupComponent,
                        SqlTableComponent,
                        SqlEditTableComponent,
                        SqlSubmitComponent
                    ]
                }]
        }] });
