// sql-components/esm2020/lib/sql.resolver.mjs
import { Injectable } from '@angular/core';
import { of, catchError } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./data.service";
export class RouteParameterResolver {
    constructor(dataService) {
        this.dataService = dataService;
    }
    resolve(route, state) {
        this.path = '';
        if (state.url !== undefined) {
            this.path = state.url;
        }
        this.r = this.dataService.pingParameters(this.path).pipe(catchError(err => {
            console.log(err);
            return of(null);
        }));
        console.log(this.r);
        this.dataService.paramSubject.next(this.r);
        this.dataService.containerSubject.next(this.r);
        return (this.r);
    }
}
RouteParameterResolver.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: RouteParameterResolver, deps: [{ token: i1.SQLDataService }], target: i0.ɵɵFactoryTarget.Injectable });
RouteParameterResolver.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: RouteParameterResolver, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: RouteParameterResolver, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }]; } });
