// sql-components/esm2020/lib/sql-delete/sql-delete.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SqlDialogComponent } from '../sql-dialog/sql-dialog.component';
import { SqlDeleteDialogComponent } from '../sql-delete-dialog/sql-delete-dialog.component';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
import * as i2 from "@angular/material/dialog";
import * as i3 from "@angular/common";
export class SqlDeleteComponent {
    constructor(_dataService, dialog) {
        this._dataService = _dataService;
        this.dialog = dialog;
        this.data = {};
        this.label = 'Delete Record';
        this.warning = 'Are you sure you want to delete this record?';
        this.class = 'btn btn-danger';
        this.style = '';
        this.change = new EventEmitter();
        this.counter = 0;
        this.myObj = this._dataService.dataSubject.subscribe(d => {
            this.data = d;
            this.counter++;
        });
    }
    handleClick(enterAnimationDuration, exitAnimationDuration) {
        this.dialog.open(SqlDeleteDialogComponent, {
            width: '350px',
            enterAnimationDuration,
            exitAnimationDuration,
        });
    }
    ngOnDestroy() {
        //    this.myObj.unsubscribe();
    }
    ngOnInit() {
    }
}
SqlDeleteComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlDeleteComponent, deps: [{ token: i1.SQLDataService }, { token: i2.MatDialog }], target: i0.ɵɵFactoryTarget.Component });
SqlDeleteComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlDeleteComponent, isStandalone: true, selector: "sql-delete", inputs: { data: "data", label: "label", warning: "warning", class: "class", style: "style" }, outputs: { change: "change" }, ngImport: i0, template: "<span *ngIf=\"data?.id!=''&&data?.id!='0'\" class=\"m-2 p-2\">\n    <button [ngClass]=\"class\" [style]=\"style\" (click)=\"handleClick('300ms', '150ms')\">{{ label }}</button> \n</span>", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlDeleteComponent, decorators: [{
            type: Component,
            args: [{ selector: 'sql-delete', standalone: true, imports: [CommonModule, SqlDialogComponent], template: "<span *ngIf=\"data?.id!=''&&data?.id!='0'\" class=\"m-2 p-2\">\n    <button [ngClass]=\"class\" [style]=\"style\" (click)=\"handleClick('300ms', '150ms')\">{{ label }}</button> \n</span>" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }, { type: i2.MatDialog }]; }, propDecorators: { data: [{
                type: Input
            }], label: [{
                type: Input
            }], warning: [{
                type: Input
            }], class: [{
                type: Input
            }], style: [{
                type: Input
            }], change: [{
                type: Output
            }] } });
