// sql-components/esm2020/lib/sql-checkbox/sql-checkbox.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { MatCheckboxModule } from '@angular/material/checkbox';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
export class SqlCheckboxComponent {
    ngAfterViewInit() {
        if (this.bs_row == 'Y') {
            this.bs_row_value = "row";
        }
        else {
            this.bs_row_value = 'none';
        }
        if (this.bs_col != '') {
            this.bs_col_value = this.bs_col;
        }
        else {
            this.bs_col_value = 'none';
        }
    }
    constructor(_dataService) {
        this._dataService = _dataService;
        this.value = '';
        this.value2 = false;
        this.fieldData = '';
        this.col = '';
        this.class = '';
        this.style = '';
        this.hint = '';
        this.icon = '';
        this.label = 'Label not set';
        this.placeholder = '';
        this.appearance = 'outline';
        this.bs_row = 'Y';
        this.bs_col = 'col-12';
        this.top_label = 'N';
        this.labelPosition = 'before';
        this.indeterminate = false;
        this.bs_row_value = 'row';
        this.bs_col_value = 'col-12';
        this.counter = 0;
        this.change = new EventEmitter();
        this.myObs = this._dataService.dataSubject.subscribe(d => {
            this.data = d;
            this.fieldData = this.data;
            this.value = this.fieldData[this.col];
            if (this.value == 'Y') {
                this.value2 = true;
            }
            else {
                this.value2 = false;
            }
            this.counter++;
        });
    }
    ngOnInit() {
    }
    handleChange() {
        this.fieldData['submit'] = 'N';
        if (this.value2 == true) {
            this.value = 'Y';
        }
        else {
            this.value = 'N';
        }
        this.fieldData[this.col] = this.value;
        this._dataService.pushNotification(this.fieldData);
    }
    ngOnDestroy() {
        //   this.myObs.unsubscribe();
    }
}
SqlCheckboxComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlCheckboxComponent, deps: [{ token: i1.SQLDataService }], target: i0.ɵɵFactoryTarget.Component });
SqlCheckboxComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlCheckboxComponent, isStandalone: true, selector: "sql-checkbox", inputs: { col: "col", data: "data", class: "class", style: "style", hint: "hint", icon: "icon", label: "label", placeholder: "placeholder", appearance: "appearance", bs_row: "bs_row", bs_col: "bs_col", top_label: "top_label", labelPosition: "labelPosition", indeterminate: "indeterminate" }, outputs: { change: "change" }, ngImport: i0, template: "<p>sql-checkbox works!</p>\n", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: FormsModule }, { kind: "ngmodule", type: MatFormFieldModule }, { kind: "ngmodule", type: MatInputModule }, { kind: "ngmodule", type: MatCheckboxModule }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlCheckboxComponent, decorators: [{
            type: Component,
            args: [{ selector: 'sql-checkbox', standalone: true, imports: [CommonModule, FormsModule, MatFormFieldModule, MatInputModule, MatCheckboxModule], template: "<p>sql-checkbox works!</p>\n" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }]; }, propDecorators: { col: [{
                type: Input
            }], data: [{
                type: Input
            }], class: [{
                type: Input
            }], style: [{
                type: Input
            }], hint: [{
                type: Input
            }], icon: [{
                type: Input
            }], label: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], appearance: [{
                type: Input
            }], bs_row: [{
                type: Input
            }], bs_col: [{
                type: Input
            }], top_label: [{
                type: Input
            }], labelPosition: [{
                type: Input
            }], indeterminate: [{
                type: Input
            }], change: [{
                type: Output
            }] } });
