// sql-components/esm2020/lib/sql-submit/sql-submit.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
import * as i2 from "@angular/common";
export class SqlSubmitComponent {
    constructor(_dataService) {
        this._dataService = _dataService;
        this.data = {};
        this.label = 'Save Changes';
        this.class = 'btn btn-primary';
        this.style = '';
        this.change = new EventEmitter();
        this.counter = 0;
        this.myObj = this._dataService.dataSubject.subscribe(d => {
            this.data = d;
            this.counter++;
        });
    }
    ngOnDestroy() {
        this.myObj.unsubscribe();
    }
    ngOnInit() {
    }
    handleClick() {
        this.data['submit'] = 'Y';
        this._dataService.pushNotification(this.data);
    }
}
SqlSubmitComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlSubmitComponent, deps: [{ token: i1.SQLDataService }], target: i0.ɵɵFactoryTarget.Component });
SqlSubmitComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlSubmitComponent, isStandalone: true, selector: "sql-submit", inputs: { data: "data", label: "label", class: "class", style: "style" }, outputs: { change: "change" }, ngImport: i0, template: "<span class=\"m-2 p-2\">\n    <button [ngClass]=\"class\" [style]=\"style\" (click)=\"handleClick()\">{{ label }}</button> \n</span>", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i2.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlSubmitComponent, decorators: [{
            type: Component,
            args: [{ selector: 'sql-submit', standalone: true, imports: [CommonModule], template: "<span class=\"m-2 p-2\">\n    <button [ngClass]=\"class\" [style]=\"style\" (click)=\"handleClick()\">{{ label }}</button> \n</span>" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }]; }, propDecorators: { data: [{
                type: Input
            }], label: [{
                type: Input
            }], class: [{
                type: Input
            }], style: [{
                type: Input
            }], change: [{
                type: Output
            }] } });
