// sql-components/esm2020/lib/sql-panel/sql-panel.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
import * as i2 from "@angular/router";
import * as i3 from "@angular/common";
export class SqlPanelComponent {
    constructor(_dataService, _router, _activatedRoute) {
        this._dataService = _dataService;
        this._router = _router;
        this._activatedRoute = _activatedRoute;
        //-- Inputs
        this.use_router = 'Y'; // does parameters come from Router.
        this.page = '';
        this.sql = ""; // query to populate panel.          
        this.subquery = ""; // subquery 1-3 go into parameters,
        this.subquery2 = "";
        this.subquery3 = "";
        this.id = '0'; // ID - ID3 go into parameters.
        this.id2 = '0';
        this.id3 = '0';
        this.card = "Y"; // Show the panel in a card Y/N
        this.closable = "N"; // Is the panel closable Y/N
        this.open = "Y"; // Does a closable form start open.
        this.class = ""; // Class for the panel container.
        this.style = ""; // Style for the panel container.
        this.title = ""; // Title of the Panel.
        this.bs_row = 'Y'; // Y means add a bootstrap row.
        this.bs_col = ''; // What bootstrap columns.
        this.change = new EventEmitter();
        this.counter = 0;
        this.parameterValue = '';
        this.parameters = { page: '', id: '', id2: '', id3: '' };
    }
    ngOnInit() {
        if (this.use_router == 'Y') {
            setTimeout(() => {
                this.myObs = this._dataService.paramSubject.subscribe(d => {
                    console.log('paramSubjct.subscribe');
                    console.log(d);
                    if (d.page !== undefined) {
                        this.page = d.page;
                    }
                    if (d.id !== undefined) {
                        this.id = d.id;
                    }
                    if (d.id2 !== undefined) {
                        this.id2 = d.id2;
                    }
                    if (d.id3 !== undefined) {
                        this.id3 = d.id3;
                    }
                    if (this.page !== 'AAA') {
                        console.log('sql: ' + this.sql);
                        console.log(d);
                        this.parameters.page = "";
                        this.parameters.id = this.id;
                        this.parameters.id2 = this.id2;
                        this.parameters.id3 = this.id3;
                        console.log(this.parameters);
                        this.myObs = this._dataService.getSQL(this.sql, this.parameters).subscribe((data) => {
                            console.log('panel init ds');
                            this.data = data[0];
                            console.log(this.data);
                            this._dataService.pageSubject.next(this.data);
                            if (this.data.id2 === undefined) {
                                this.data.id2 = '';
                            }
                            if (this.data.id3 === undefined) {
                                this.data.id3 = '';
                            }
                            this._dataService.containerSubject.next(this.data);
                        });
                    }
                });
            }, 500);
        }
    }
    ngAfterViewInit() {
    }
    ngDoCheck() {
    }
    ngOnChanges(changes) {
    }
    ngOnDestroy() {
        //  this.myObs.unsubscribe();
    }
}
SqlPanelComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlPanelComponent, deps: [{ token: i1.SQLDataService }, { token: i2.Router }, { token: i2.ActivatedRoute }], target: i0.ɵɵFactoryTarget.Component });
SqlPanelComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlPanelComponent, isStandalone: true, selector: "sql-panel", inputs: { use_router: "use_router", page: "page", data: "data", sql: "sql", subquery: "subquery", subquery2: "subquery2", subquery3: "subquery3", id: "id", id2: "id2", id3: "id3", card: "card", closable: "closable", open: "open", class: "class", style: "style", title: "title", bs_row: "bs_row", bs_col: "bs_col" }, outputs: { change: "change" }, usesOnChanges: true, ngImport: i0, template: "<ng-container *ngIf=\"card=='N'\">\n    <div *ngIf=\"bs_row=='Y'\" class=\"row\">\n        <div [class]=\"bs_col\">\n            <ng-container *ngTemplateOutlet=\"fullTmpl\"></ng-container>\n        </div>\n    </div>\n    <!-- bs_row == 'N' and bs_col exists -->\n    <div *ngIf=\"bs_row!='Y'&&bs_col!=''\">\n        <div [class]=\"bs_col\">\n            <ng-container *ngTemplateOutlet=\"fullTmpl\"></ng-container>\n        </div>\n    </div>\n    <!-- bs_row == 'N' and bs_col exists -->\n    <div *ngIf=\"bs_row!='Y'&&bs_col==''\">\n            <ng-container *ngTemplateOutlet=\"fullTmpl\"></ng-container>\n    </div>\n</ng-container>\n\n<!-- it is a card -->\n<div *ngIf=\"card=='Y'\">\n    <div *ngIf=\"bs_row=='Y'\" class=\"row\">\n        <div [class]=\"bs_col\">\n            <div class=\"card\">\n                <div class=\"card-header\" style=\"margin-top: 0px;\">\n                    <h2 *ngIf=\"title!=''\" p-2>{{ title }}</h2>\n                    <ng-container *ngTemplateOutlet=\"headerTmpl\"></ng-container>\n                </div>\n                <div class=\"card-body\">\n                    <ng-container *ngTemplateOutlet=\"footerTmpl\"></ng-container>\n                </div>\n            </div>\n        </div>\n    </div>\n    <div *ngIf=\"bs_row!='Y'&&bs_col!=''\">\n        <div [class]=\"bs_col\">\n            <div class=\"card\">\n                <div class=\"card-header p-2\">\n                    <h2 *ngIf=\"title!=''\" p-2>{{ title }}</h2>\n                    <ng-container *ngTemplateOutlet=\"headerTmpl\"></ng-container>\n                </div>\n                <div class=\"card-body\">\n                    <ng-container *ngTemplateOutlet=\"footerTmpl\"></ng-container>\n                </div>\n            </div>\n        </div>\n    </div>    \n    <div *ngIf=\"bs_row!='Y'&&bs_col==''\">\n            <div class=\"card\">\n                <div class=\"card-header\">\n                    <h2 *ngIf=\"title!=''\" p-2>{{ title }}</h2>\n                    <ng-container *ngTemplateOutlet=\"headerTmpl\"></ng-container>\n                </div>\n                <div class=\"card-body\">\n                    <ng-container *ngTemplateOutlet=\"footerTmpl\"></ng-container>\n                </div>\n            </div>\n    </div>    \n</div>\n\n<ng-template #footerTmpl>\n    <ng-content select=\"[footer]\"></ng-content>\n</ng-template>\n\n<ng-template #headerTmpl>\n    <ng-content select=\"[header]\"></ng-content>\n</ng-template>\n\n<ng-template #fullTmpl>\n    <ng-content></ng-content>\n</ng-template>\n", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i3.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "ngmodule", type: RouterModule }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlPanelComponent, decorators: [{
            type: Component,
            args: [{ selector: 'sql-panel', standalone: true, imports: [CommonModule, RouterModule], template: "<ng-container *ngIf=\"card=='N'\">\n    <div *ngIf=\"bs_row=='Y'\" class=\"row\">\n        <div [class]=\"bs_col\">\n            <ng-container *ngTemplateOutlet=\"fullTmpl\"></ng-container>\n        </div>\n    </div>\n    <!-- bs_row == 'N' and bs_col exists -->\n    <div *ngIf=\"bs_row!='Y'&&bs_col!=''\">\n        <div [class]=\"bs_col\">\n            <ng-container *ngTemplateOutlet=\"fullTmpl\"></ng-container>\n        </div>\n    </div>\n    <!-- bs_row == 'N' and bs_col exists -->\n    <div *ngIf=\"bs_row!='Y'&&bs_col==''\">\n            <ng-container *ngTemplateOutlet=\"fullTmpl\"></ng-container>\n    </div>\n</ng-container>\n\n<!-- it is a card -->\n<div *ngIf=\"card=='Y'\">\n    <div *ngIf=\"bs_row=='Y'\" class=\"row\">\n        <div [class]=\"bs_col\">\n            <div class=\"card\">\n                <div class=\"card-header\" style=\"margin-top: 0px;\">\n                    <h2 *ngIf=\"title!=''\" p-2>{{ title }}</h2>\n                    <ng-container *ngTemplateOutlet=\"headerTmpl\"></ng-container>\n                </div>\n                <div class=\"card-body\">\n                    <ng-container *ngTemplateOutlet=\"footerTmpl\"></ng-container>\n                </div>\n            </div>\n        </div>\n    </div>\n    <div *ngIf=\"bs_row!='Y'&&bs_col!=''\">\n        <div [class]=\"bs_col\">\n            <div class=\"card\">\n                <div class=\"card-header p-2\">\n                    <h2 *ngIf=\"title!=''\" p-2>{{ title }}</h2>\n                    <ng-container *ngTemplateOutlet=\"headerTmpl\"></ng-container>\n                </div>\n                <div class=\"card-body\">\n                    <ng-container *ngTemplateOutlet=\"footerTmpl\"></ng-container>\n                </div>\n            </div>\n        </div>\n    </div>    \n    <div *ngIf=\"bs_row!='Y'&&bs_col==''\">\n            <div class=\"card\">\n                <div class=\"card-header\">\n                    <h2 *ngIf=\"title!=''\" p-2>{{ title }}</h2>\n                    <ng-container *ngTemplateOutlet=\"headerTmpl\"></ng-container>\n                </div>\n                <div class=\"card-body\">\n                    <ng-container *ngTemplateOutlet=\"footerTmpl\"></ng-container>\n                </div>\n            </div>\n    </div>    \n</div>\n\n<ng-template #footerTmpl>\n    <ng-content select=\"[footer]\"></ng-content>\n</ng-template>\n\n<ng-template #headerTmpl>\n    <ng-content select=\"[header]\"></ng-content>\n</ng-template>\n\n<ng-template #fullTmpl>\n    <ng-content></ng-content>\n</ng-template>\n" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }, { type: i2.Router }, { type: i2.ActivatedRoute }]; }, propDecorators: { use_router: [{
                type: Input
            }], page: [{
                type: Input
            }], data: [{
                type: Input
            }], sql: [{
                type: Input
            }], subquery: [{
                type: Input
            }], subquery2: [{
                type: Input
            }], subquery3: [{
                type: Input
            }], id: [{
                type: Input
            }], id2: [{
                type: Input
            }], id3: [{
                type: Input
            }], card: [{
                type: Input
            }], closable: [{
                type: Input
            }], open: [{
                type: Input
            }], class: [{
                type: Input
            }], style: [{
                type: Input
            }], title: [{
                type: Input
            }], bs_row: [{
                type: Input
            }], bs_col: [{
                type: Input
            }], change: [{
                type: Output
            }] } });
