// sql-components/esm2020/lib/sql-delete-dialog/sql-delete-dialog.component.mjs
import { Component, Input, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatDialogModule } from '@angular/material/dialog';
import { MatButtonModule } from '@angular/material/button';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
import * as i2 from "@angular/material/dialog";
export class SqlDeleteDialogComponent {
    constructor(_dataService, dialog) {
        this._dataService = _dataService;
        this.dialog = dialog;
        this.data = {};
        this.change = new EventEmitter();
        this.myObj = this._dataService.dataSubject.subscribe(d => {
            this.data = d;
        });
    }
    ngOnDestroy() {
        this.myObj.unsubscribe();
    }
    ngOnInit() {
    }
    handleClick() {
        this.data['submit'] = 'D';
        this._dataService.pushNotification(this.data);
    }
}
SqlDeleteDialogComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlDeleteDialogComponent, deps: [{ token: i1.SQLDataService }, { token: i2.MatDialog }], target: i0.ɵɵFactoryTarget.Component });
SqlDeleteDialogComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlDeleteDialogComponent, isStandalone: true, selector: "lib-sql-delete-dialog", inputs: { data: "data" }, ngImport: i0, template: "<h1 mat-dialog-title>Delete Record</h1>\n<div mat-dialog-content>\n  Are you sure to want to delete this record?  This cannot be reversed.\n</div>\n<div mat-dialog-actions>\n  <button class=\"btn btn-primary\" mat-dialog-close>No</button>\n  <button style=\"margin-left: 20px;\" (click)=\"handleClick()\" class=\"btn btn-danger\" mat-dialog-close cdkFocusInitial>Yes</button>\n</div>", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatDialogModule }, { kind: "directive", type: i2.MatDialogClose, selector: "[mat-dialog-close], [matDialogClose]", inputs: ["aria-label", "type", "mat-dialog-close", "matDialogClose"], exportAs: ["matDialogClose"] }, { kind: "directive", type: i2.MatDialogTitle, selector: "[mat-dialog-title], [matDialogTitle]", inputs: ["id"], exportAs: ["matDialogTitle"] }, { kind: "directive", type: i2.MatDialogContent, selector: "[mat-dialog-content], mat-dialog-content, [matDialogContent]" }, { kind: "directive", type: i2.MatDialogActions, selector: "[mat-dialog-actions], mat-dialog-actions, [matDialogActions]", inputs: ["align"] }, { kind: "ngmodule", type: MatButtonModule }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlDeleteDialogComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-sql-delete-dialog', standalone: true, imports: [CommonModule, MatDialogModule, MatButtonModule], template: "<h1 mat-dialog-title>Delete Record</h1>\n<div mat-dialog-content>\n  Are you sure to want to delete this record?  This cannot be reversed.\n</div>\n<div mat-dialog-actions>\n  <button class=\"btn btn-primary\" mat-dialog-close>No</button>\n  <button style=\"margin-left: 20px;\" (click)=\"handleClick()\" class=\"btn btn-danger\" mat-dialog-close cdkFocusInitial>Yes</button>\n</div>" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }, { type: i2.MatDialog }]; }, propDecorators: { data: [{
                type: Input
            }] } });
