// ngx-table-pagination/esm2015/lib/pagination.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { PaginationComponent } from './pagination.component';
import { PaginationService } from './pagination.service';
import { PaginationDirective } from './pagination.directive';
import { PaginationPipe } from './pagination.pipe';
export { PaginationService } from './pagination.service';
export { PaginationComponent } from './pagination.component';
export { PaginationDirective } from './pagination.directive';
export { PaginationPipe } from './pagination.pipe';
export class NgxTablePaginationModule {
}
NgxTablePaginationModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    PaginationComponent,
                    PaginationPipe,
                    PaginationDirective
                ],
                imports: [
                    CommonModule,
                    FormsModule
                ],
                providers: [PaginationService],
                exports: [PaginationComponent,
                    PaginationPipe,
                    PaginationDirective]
            },] }
];
