// sql-components/esm2020/lib/sql-default-value/sql-default-value.component.mjs
import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
export class SqlDefaultValueComponent {
    constructor(_dataService) {
        this._dataService = _dataService;
        this.col = '';
        this.value = '';
        this.myObs = this._dataService.dataSubject.subscribe(d => {
            this.data = d;
        });
    }
    ngAfterViewInit() {
        console.log('sql-default-value');
        console.log(this.value);
        console.log(this.col);
        setTimeout(() => {
            this.data[this.col] = this.value;
            console.log('sql-default-value: timer');
            console.log(this.data);
            this._dataService.dataSubject.next(this.data);
        }, 1000);
    }
    ngOnDestroy() {
        this.myObs.unsubscribe();
    }
}
SqlDefaultValueComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlDefaultValueComponent, deps: [{ token: i1.SQLDataService }], target: i0.ɵɵFactoryTarget.Component });
SqlDefaultValueComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: SqlDefaultValueComponent, isStandalone: true, selector: "sql-default-value", inputs: { col: "col", value: "value" }, ngImport: i0, template: "", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: SqlDefaultValueComponent, decorators: [{
            type: Component,
            args: [{ selector: 'sql-default-value', standalone: true, imports: [CommonModule], template: "" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }]; }, propDecorators: { col: [{
                type: Input
            }], value: [{
                type: Input
            }] } });
