// sql-components/esm2020/lib/pre-insert-trigger/pre-insert-trigger.component.mjs
import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "../data.service";
export class PreInsertTriggerComponent {
    constructor(_dataService) {
        this._dataService = _dataService;
        this.name = 'pre-insert';
        this.sql = '';
        this.order = '0';
    }
    ngOnDestroy() {
        this.myObs.unsubscribe();
    }
    ngOnChanges(changes) {
    }
    registerService() {
        let count = 0;
        this.myObs = this._dataService.dataSubject.subscribe(d => {
            this.data = d;
            if (this.data.triggers === undefined) {
                let t = [];
                this.data.triggers = t;
            }
            count = 0;
            this.data.triggers.forEach((element) => {
                if (element.type == 'pre-insert-trigger' && element.name == this.name) {
                    count++;
                }
            });
            if (count == 0) {
                let tr = { type: 'pre-insert-trigger', name: this.name, sql: this.sql, order: this.order };
                this.data.triggers.push(tr);
                this._dataService.dataSubject.next(this.data);
            }
        });
    }
    ngAfterViewInit() {
        setTimeout(() => {
            this.registerService();
        }, 1000);
    }
}
PreInsertTriggerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: PreInsertTriggerComponent, deps: [{ token: i1.SQLDataService }], target: i0.ɵɵFactoryTarget.Component });
PreInsertTriggerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.2.3", type: PreInsertTriggerComponent, isStandalone: true, selector: "pre-insert-trigger", inputs: { name: "name", sql: "sql", order: "order" }, usesOnChanges: true, ngImport: i0, template: "", styles: [""], dependencies: [{ kind: "ngmodule", type: CommonModule }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.3", ngImport: i0, type: PreInsertTriggerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'pre-insert-trigger', standalone: true, imports: [CommonModule], template: "" }]
        }], ctorParameters: function () { return [{ type: i1.SQLDataService }]; }, propDecorators: { name: [{
                type: Input
            }], sql: [{
                type: Input
            }], order: [{
                type: Input
            }] } });
